rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,204 @@
+![Element-First Spacing](./images/spatial/element-first.png)
+
+- **Content first (strict internal padding):** This is to be used when the content is dynamic or less predictable. Therefore, we will enforce the strict internal padding to those components. Simply put, this is the spacing around the elements.
+![Content-First Spacing](./images/spatial/content-first.png)
+
+### **Spatial Scale**
+We'll follow a **8pt** linear scale for the elements with the **4pt** half-step for spacing smaller content (icons, etc.) and typography (line-heights).
+
+This is how the scale will look like:
+
+| Alias      | Value |
+|------------|-------|
+| $spacer-0  | 0px   |
+| $spacer-1  | 4px   |
+| $spacer-2  | 8px   |
+| $spacer-3  | 12px  |
+| $spacer-4  | 16px  |
+| $spacer-5  | 20px  |
+| $spacer-6  | 24px  |
+| $spacer-7  | 32px  |
+| $spacer-8  | 40px  |
+| $spacer-9  | 48px  |
+| $spacer-10 | 56px  |
+
+>These are mixed values, it contains multiple of `4` as well as `8`.
+
+### **Grids**
+The grid system helps align page elements based on sequenced columns and rows. Basically it helps you arrange your structure into structured manner. Here we'll look into 2D organisation of information.
+
+### **Column Grid**
+As the name suggests, this type of grid will help you organise content into evenly spaced vertical columns. The spacer which divide this grid is called as `gutter`. We'll use a **12 column grid** as it helps divide the area into *halfs*, *thirds*, *fourths*, *sixths*.
+
+Here we'll have 3 types of column grids, these will be used depending on the use case. All of them are following the same maximum width of `1280px`.
+1. **Default:** This has a gutter of `32px` & will be used on the internal dashboard.
+![Default Column-Grid](./images/spatial/col-grid-default.png)
+
+2. **Condensed:** This has a gutter of `16px` & will be used on mobile layouts *(only for dashboards)*.
+![Condensed Column-Grid](./images/spatial/col-grid-condensed.png)
+
+3. **Spacious:** This has a gutter of `48px` & will be used for landing/marketing pages *(on desktop)*.
+![Condensed Column-Grid](./images/spatial/col-grid-spacious.png)
+
+These column grids can also be used internally based on the what component we are building. For example, let's say we are building inline form, then we'll use condensed col-grid for organising elements.
+
+### **Marketing Pages**
+For marketing pages, the UI is more spacious. Therefore, we can go ahead and use the **spacious system** (gutter of `48px`) and on mobiles the same system will be converterd to a default system (gutter of `24px`).
+### **Columns, Gutters and Margins**
+A place where the content is placed inside a layout. These vary depending on the screen size or device you are designing on. Here are few examples,
+
+- **Mobile Layout**
+![Mobile Layout](./images/spatial/mobile-layout.png)
+
+- **Tablet Layout**
+![Tablet Layout](./images/spatial/tablet-layout.png)
+
+- **Desktop Layout**
```

Comment:
Why are we using condensed layout grid for desktops? In the above section we said that this will only be used on mobile right?

Reply:
@kamleshchandnani Actually, we'll use mic of both condensed and default. Only one system cannot fulfil all the needs.
For example, we have the main layout which has a sidebar and a dashboard content window. Now the gutter between these can follow `default` system but when we the segregation happens inside the sidebar/content view we will use the `condensed` system.
> How should I convey this better?

Reply:
I saw the notes you added for this. It conveys things better now 👍